sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -508,164 +498,202 @@ export class EventHubSender extends LinkEntity {
     message: AmqpMessage | Buffer,
     options: SendOptions & EventHubProducerOptions = {}
   ): Promise<void> {
-
     const abortSignal: AbortSignalLike | undefined = options.abortSignal;
-    const sendEventPromise = () =>
+
+    const sendOperationPromise = () =>
```

Comment:
There are 2 functions creating promises. The outer function `sendOperationPromise` is what is passed to `retry`.
Both initialization of sender link and the actual send is happening in the inner function `sendEventPromise`. But event handlers for the actual send is in the outer promise. For better readability, I would expect all actual send related code to be together.

Any reason we could not do that keep only the timer on the outer promise?

Reply:
1 reason I can think of for the current design is that you need the timer to be in outer promise. On timeout, we are to clear the listeners on the sender. Therefore, the listeners were added on the outer promise.
Is that so?

Reply:
Yes, needed access to removeListeners as part of timeout handling. Hence moved them all to one place

Reply:
Have you tried initializing the listeners to do a no-op in the outer promise and then set them to the actual listeners in the inner promise?

The problem with the current design is with the separation of concerns.
When the actual send succeeds, the outer promise resolves, inner promise doesnt.
But, since the inner promise is the one that is actually doing the work of sending the event, one would expect the inner promise to resolve/reject based on whether the actual send was successful or not

Reply:
Discussed offline, the resolution of promises is not affected as the listeners are on sender link and not tied with promises as such.
However, we agreed to move around callbacks/variable definitions to be within `sendEvents` as much as possible. 
But since recent decision about separating out 'send' timeout to be overall timeout minus time taken for 'init' for separation of concerns (in case of management requests, looking to clarify if we wanted to do same here as well). Thoughts?
@ramya-rao-a @chradek

Reply:
Discussed offline, the timeout calculation separation was specific to `managementRequest` and current simplified changes are fine~

Reply:
Well, actually one thing I just noticed is that you don't set the timeout until after you initialize the link. You'll want to move it up so it applies to the whole operation.

Reply:
Yes, just updated it. (Sorry our goal to simplify changes couldn't be met! :-/ other callback definitions do need to be moved around)